code_samples/api/rest_api/create_image.xml.php

```diff
@@ -66,94 +71,68 @@
+        ],
+        'body' => $data,
+    ]);
+} catch (HttpException\TransportExceptionInterface $exception) {
+    echo "Client error: {$exception->getMessage()}\n";
     exit(3);
 }
 
-if (201 !== $responseCode = curl_getinfo($curl, CURLINFO_RESPONSE_CODE)) {
-    if (!empty($response) && $doc->loadXML($response) && 'ErrorMessage' === $doc->firstChild->nodeName) {
+if (201 !== $responseCode = $response->getStatusCode()) {
+    if (!empty($response->getContent(false)) && @$doc->loadXML($response->getContent(false)) && 'ErrorMessage' === $doc->firstChild->nodeName) {
         echo "Server error: {$doc->getElementsByTagName('errorCode')->item(0)->nodeValue} {$doc->getElementsByTagName('errorMessage')->item(0)->nodeValue}\n";
         echo "\t{$doc->getElementsByTagName('errorDescription')->item(0)->nodeValue}\n";
-        curl_close($curl);
         exit(4);
     }
+    $responseHeaders = $response->getInfo('response_headers');
     $error = $responseHeaders[0] ?? $responseCode;
     echo "Server error: $error\n";
-    curl_close($curl);
     exit(5);
 }
 
-$doc->loadXML($response);
+$doc->loadXML($response->getContent());
 
 if ('Content' !== $doc->firstChild->nodeName || !$doc->firstChild->hasAttribute('id')) {
     echo "Response error: Unexpected response structure\n";
-    curl_close($curl);
     exit(6);
 }
 
 $contentId = $doc->firstChild->getAttribute('id');
 
-curl_reset($curl);
-$responseHeaders = [];
-
-curl_setopt_array($curl, [
-    CURLOPT_USERPWD => "$username:$password",
-    CURLOPT_URL => "$baseUrl/content/objects/$contentId/versions/1",
-    CURLOPT_POST => true,
-    CURLOPT_HTTPHEADER => [
-        'X-HTTP-Method-Override: PUBLISH',
-        'Accept: application/json',
-    ],
-    CURLOPT_RETURNTRANSFER => true,
-]);
-
-$response = curl_exec($curl);
-
-if ($error = curl_error($curl)) {
-    echo "cURL error: $error\n";
-    curl_close($curl);
+try {
+    $response = $client->request('PUBLISH', "$baseUrl/content/objects/$contentId/versions/1", [
+        'headers' => [
+            'Accept: application/xml',
+        ],
+    ]);
+} catch (HttpException\TransportExceptionInterface $exception) {
+    echo "Client error: {$exception->getMessage()}\n";
     exit(7);
 }
 
-if (204 !== $responseCode = curl_getinfo($curl, CURLINFO_RESPONSE_CODE)) {
-    if (!empty($response) && $doc->loadXML($response) && 'ErrorMessage' === $doc->firstChild->nodeName) {
+if (204 !== $responseCode = $response->getStatusCode()) {
+    if (!empty($response->getContent(false)) && @$doc->loadXML($response->getContent(false)) && 'ErrorMessage' === $doc->firstChild->nodeName) {
         echo "Server error: {$doc->getElementsByTagName('errorCode')->item(0)->nodeValue} {{$doc->getElementsByTagName('errorMessage')->item(0)->nodeValue}\n";
         echo "\t{$doc->getElementsByTagName('errorDescription')->item(0)->nodeValue}\n";
-        curl_close($curl);
         exit(8);
     }
+    $responseHeaders = $response->getInfo('response_headers');
```

Comment:
Can't we rely on `$response->getHeaders()`, can we?

Reply:
`$response->getHeaders()` doesn't give the status line (the first line). Here, the goal is to have the status message along side the status code.